Graceful shutdown during podman kube down

andremarianiello opened the pull request:
Fixes #22397
<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]". That will prevent functional tests from running and save time and energy.

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?
Yes
<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
Fixed a bug where `podman kube down` would not respect the StopTimeout and StopSignal of containers that it stopped
```

andremarianiello commented:
`podman pod stop` handles the signalling correctly by using `Timeout: -1` to respect the settings of the individual containers, so I copied that behavior here.

rhatdan commented:
Thanks @andremarianiello 
LGTM

mheon commented:
Code LGTM. Looks like this broke one of the `play kube` tests, which has a container that doesn't respond to SIGTERM. Could set that container's stop signal to SIGKILL or just do a `podman stop --all -t 0` before doing the `podman kube down`

Luap99 commented:
Can you add a proper regression test please? This seems very easy to regress on later.

You can run something like `sh -c "trap 'touch /test/somefile; exit' SIGTERM; echo READY; while :; do sleep 0.1; done"` in the container where you mount some external dir onto /test and the after kube down make sure somefile was created in the directory.

andremarianiello commented:
I updated the newly failing test `with named volume subpaths` to terminate on SIGTERM, and I added a new test per @Luap99 's suggestion to test that `kube down` sends SIGTERM using a signal handler that writes to a mounted file. Please let me know if there is anything else I should address before this gets merged. Thanks!

andremarianiello commented:
It's not clear to me why the rpm-build and testing-farm checks are failing. I think I need to rely on someone more knowledgeable to check on that.

Luap99 commented:
> It's not clear to me why the rpm-build and testing-farm checks are failing. I think I need to rely on someone more knowledgeable to check on that.

you can ignore the packit failures, they are not required

andremarianiello commented:
@Luap99 Thanks!

Luap99 commented on test/e2e/play_kube_test.go:
```diff
@@ -5568,6 +5595,28 @@ spec:
 	})
 
+	It("with graceful shutdown", func() {
+
```
nit: unnecessary newline

Luap99 commented on test/e2e/play_kube_test.go:
```diff
@@ -5568,6 +5595,28 @@ spec:
+		teardown.WaitWithDefaultTimeout()
+		Expect(teardown).Should(ExitCleanly())
+
+		session := podmanTest.Podman([]string{"run", "--volume", "testvol:/testvol", CITEST_IMAGE, "sh", "-c", "cat /testvol/termfile"})
```
nit: can just call `"cat", "file"` without the `sh -c` part

Luap99 reviewed (approved):
LGTM, thanks. Some minor nits not worth to repush for.

Luap99 commented:
/lgtm
